### ISSUE TITLE
Association Path isn't working in OrderBy clause

### ISSUE THREAD

**Issue by Goldmensch** [micronaut-projects/micronaut-data#1521]
### Expected Behavior

The query should contain a valid order by clause.

### Actual Behaviour

The compilation is failing, because Micronaut can't find the property: 
```
Unable to implement Repository method: QuestionRepo.findByIdTemplateIdAndIdNumberGreaterThanEqualsOrderByIdNumber(int internalId,int number). Cannot order by non-existent property: idNumber
  List<Question> findByIdTemplateIdAndIdNumberGreaterThanEqualsOrderByIdNumber(int internalId,
                 ^
```
My guess is, that it's due to the CamelCaseRegex
https://regex101.com/r/aMudnD/1 (IdNumber at the end)

### Version

3.4.2

**Comment by radovanradic** [comment]
@Goldmensch can you please show Question mapping entity?

**Comment by Goldmensch** [comment]
> @Goldmensch can you please show Question mapping entity?

It's the same as in #1628.

**Comment by radovanradic** [comment]
Will be fixed in micronaut-data 3.7.3
